test-tests/shared/src/test/scala/zio/test/GenSpec.scala

```diff
@@ -1,16 +1,20 @@
 package zio.test
 
+import java.time.{ Instant, LocalDateTime, OffsetDateTime, ZoneOffset }
+
 import scala.math.Numeric.DoubleIsFractional
 
+import zio.ZIO
+import zio.duration._
 import zio.random.Random
-import zio.stream.ZStream
 import zio.test.Assertion._
+import zio.test.GenUtils._
 import zio.test.TestAspect.scala2Only
-import zio.test.environment.TestRandom
 import zio.test.{ check => Check, checkN => CheckN }
-import zio.{ Managed, UIO, ZIO }
 
 object GenSpec extends ZIOBaseSpec {
+  implicit val localDateTimeOrdering: Ordering[LocalDateTime] = _ compareTo _
```

Comment:
Looked into this a bit more. It looks like this implicit is actually available on Scala 2.13.1 but not on 2.12.10. Specifically, they use a slightly different encoding to derive an `Ordering` for any type for which `Comparable` is defined (which includes `LocalDateTime`). In Scala 2.13:

```scala
type AsComparable[A] = A => Comparable[_ >: A]

implicit def ordered[A](implicit asComparable: AsComparable[A]): Ordering[A] = new Ordering[A] {
  def compare(x: A, y: A): Int = asComparable(x).compareTo(y)
}
```

In Scala 2.12:

```scala
implicit def ordered[A <% Comparable[A]]: Ordering[A] = new Ordering[A] {
  def compare(x: A, y: A): Int = x compareTo y
}
```

For some reason the view bound in Scala 2.12 doesn't work, possibly because `A` is constrained to be the same type versus a super type as in Scala 2.13 and `LocalDateTime` actually implements `Comparable[ChronoLocalDateTime`.

We could potentially create a compatibility trait that would provide this instance but given that it is only used in this file I don't think it is really necessary.

Reply:
Thanks - I guess it's a nice improvement for 2.13 :) 

I did find the 2.12 definition and guessed the reason it wasn't working was because the A type didn't line up. It looks like OffsetDateTime directly implements `compareTo` while LocalDateTime overrides `compareTo` on in `ChronoLocalDateTime`, but since `ChronoLocalDateTime` implements `compareTo` I would have thought it should still be considered a `Comparable`.

I'll throw that type signature (`type AsWhatever[A] = A => Whatever[_ >: A]`)  in my bag of tricks - looks handy for these sort of circumstances!